jenkins/Jenkinsfile

```diff
@@ -0,0 +1,240 @@
+
+  def currLine = ""
+  for (String currReplacement in replacementMap.keySet()) {
+    for (int j = 0; j < contents.size(); j++) {
+      currLine = contents[j]
+      if (currLine.contains(currReplacement))  {
+        location = currLine.indexOf(currReplacement)
+        //find the image name from imageDirs.txt in the config, and substring it so we preserve everything up to the hash
+        //we add 1 because the image name is suffixed with a ':'' before the hash that we wish to replace
+        //note we must also close the quote with this approach
+        def String newLine = currLine.substring(0, location + currReplacement.length() + 1) + replacementMap[currReplacement] + '"'
+        println("replacing line $currLine with $newLine")
+        contents[j] = newLine
+      }
+    }
+  }
+
+  println("contents of $filePath after modification is $contents")
+  def joinedContents = contents.join('\n')
+  def workingDir = pwd()
+ 
+  fp = createFilePath("$workingDir/$filePath")
+  //we add a trailing newline to the file as its lost in the split/join process
+  fp.write(joinedContents + "\n", "UTF-8")
+}
+
+//Filepath is an object that abstracts away the fact the file can be on a remote jenkins node or the master jenkins node
+def FilePath createFilePath(String path) {
+    if (env['NODE_NAME'] == null) {
+        throw new RuntimeException("envvar NODE_NAME is not set, probably not inside a node {} or running a different version of Jenkins!");
+    } else if (env['NODE_NAME'].equals("master")) {
+        return new FilePath(path);
+    } else {
+        println("detecting remote file in createFilePath")
+        return new FilePath(Jenkins.getInstance().getComputer(env['NODE_NAME']).getChannel(), path);
+    }
+}
+
+pipeline {
+   agent { label 'node210' }
+
+  options { 
+    //we don't want to be building more than one PR's changes at once, we will queue them instead
+    disableConcurrentBuilds() 
+  }
+
+  triggers {
+       // poll repo every 2 minute for changes
+       pollSCM('*/2 * * * *')
+  }
+
+   stages {
+     
+    stage('Terra Docker Git') {
+      steps {
+        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/terra-docker.git', branch: 'master'
+      }
+    }
+
+    stage('Docker Auth') {
+      steps {
+        //the chown is necessary because something on the jenkins instance keeps touching the logs with sudo and changing the owner to root
+        //this which causes gcloud auth command to fail
+        sh """
+          sudo chown -R jenkins:jenkins /home/jenkins/.config
+          docker run --rm  -v /etc/vault-token-dsde:/root/.vault-token:ro broadinstitute/dsde-toolbox:latest vault read --format=json secret/dsde/dsp-techops/common/dspci-wb-gcr-service-account.json | jq .data > dspci-wb-gcr-service-account.json   
+          gcloud auth activate-service-account --key-file=dspci-wb-gcr-service-account.json
+          gcloud auth configure-docker --quiet
```

Comment:
we probably need to fix this job to the same node we run welder publish image cuz `gcloud auth configure-docker` causes trouble for other jenkins jobs

Reply:
the 'agent' line at the top of the file does this. I'll add a comment